assertion/function/assertiontree/contract.go

```diff
@@ -461,3 +460,17 @@ func guardExpr(rootNode *RootAssertionNode, expr TrackableExpr, guard util.Guard
 				lookedUpNode.ConsumeTriggers(), guard))
 	}
 }
+
+// extractLHSRHS extracts the left-hand side and right-hand side of an assignment statement or a variable declaration
+func extractLHSRHS(node ast.Node) (lhs, rhs []ast.Expr) {
+	switch expr := node.(type) {
+	case *ast.AssignStmt:
+		lhs, rhs = expr.Lhs, expr.Rhs
+	case *ast.ValueSpec:
+		for _, name := range expr.Names {
+			lhs = append(lhs, name)
+		}
+		rhs = expr.Values
+	}
+	return
+}
```

Comment:
As we discussed internally, let's make this a function in `asthelper` since I feel this is general enough to warrant being there :)

Reply:
Done :)